src/main/java/org/testng/GuiceHelper.java

```diff
@@ -121,12 +122,12 @@ public static Injector createInjector(ITestContext context,
     Class<? extends IModuleFactory> factory = guice.moduleFactory();
     if (factory != IModuleFactory.class) {
       IModuleFactory factoryInstance = parentInjector.getInstance(factory);
-      Module moduleClass = factoryInstance.createModule(context, testClass);
-      if (moduleClass != null) {
-        result.add(moduleClass);
+      Module module = factoryInstance.createModule(context, testClass);
+      if (module != null) {
+        result.add(module);
       }
     }
-
+    ServiceLoader.load(IModule.class).forEach(result::add);
```

Comment:
What would happen when the same module is being loaded multiple times because of multiple SPI references in the CLASSPATH ?

Reply:
It depends on the module itself. In case of conflicts a redundant dependency should be excluded. It seems correct to me.

Reply:
@bartoszpop - I am not that familiar with Guice DI, but the current implementation doesn't even provide a way in which some SPI wiring can be skipped. 

> In case of conflicts a redundant dependency should be excluded.

Who would be doing that ?

Reply:
Sorry, a module will be loaded only once.

Reply:
How would we ensure that ? Are you saying that Guice ensures that because we seem to be adding to a List right ?

Reply:
I mean that if you have the same class implementing the IModule interface (eg. com.example.SomeModule) listed in two separate SPI files (some.jar/META-INF/services/org.testng.IModule and other.jar/META-INF/services/org.testng.IModule), then java.util.ServiceLoader#load will return a list with a single instance of this module (not two instances of com.example.SomeModule).

Reply:
Ok. Thanks. I didn't know that. Good learning.